### DISCUSSION TITLE
Add support for the NGINX HTTP3/QUIC module

### DISCUSSION THREAD

**Discussion by brianehlert** [nginx/kubernetes-ingress#4240] [Ideas]
NGINX has recently introduced HTTP3/QUIC support with NGINX Plus.
https://www.nginx.com/blog/nginx-plus-r30-released/#New-Features-in-Detail

It is optional for NGINX: https://nginx.org/en/docs/http/ngx_http_v3_module.html

HTTP3 is a combination of listener and server settings, not unlike MQTT or http2.

Should we consider implementing http3 for NGINX Plus only?
Should we also implement http3 for the free edition of this project?

In both cases I am thinking the implementation should progress in this way:

- VirtualServer (or similar CRD)
- Annotations based on demand

**Comment by jasonwilliams14** [comment]
Additional specific items for `HTTP3/QUIC`:
Below are the new directives introduced by NGINX to support HTTP3/QUIC. In addition to the new directives that are introduced, TLS1.3 is required for QUIC to work correctly.


New directives and documentation link:

[NGINX QUIC](https://nginx.org/en/docs/http/ngx_http_v3_module.html)

Enables http3/quic

http3
```
http3 on/off;
Default: on
Context: http/server
```


http3_hq on | off;

```
Default:	http3_hq off;
Context:	http, server
Enables HTTP/0.9 protocol negotiation used in 
```

http3_max_concurrent_streams number;
```
Default:	http3_max_concurrent_streams 128;
Context:	http, server

```
`http3_stream_buffer_size size;`
```
Default:	http3_stream_buffer_size 64k;
Context:	http, server
```

quic_active_connection_id_limit number;
```
Default:	
quic_active_connection_id_limit 2;
Context:	http, server
```

quic_retry on | off;
```
Default:	
quic_retry off;
Context:	http, server
```

One new variable has been also introduced as part of `http3/quic` allowing the extending of the logging to capture `http3` connections in NGINX logs.

```
$http3
```

**Comment by jjngx** [comment]
@shaun-nx do we need the QA still here? Is this captured somewhere else?
